### PR TITLE
init androidenv.buildGradleApp

### PR DESCRIPTION
**Pull request by matthewbauer**
###### Motivation for this change

This is the last big piece of code that is used by [reflex-platform](https://github.com/reflex-frp/reflex-platform) but missing in nixpkgs. It is used to generate apps using the gradle build system.

Need feedback on whether this is useful in Nixpkgs.

/cc @ali-abrar @ryantrinkle @Ericson2314

### CODE REVIEW

**Comment by matthewbauer** [comment]
Some of these commits need to be reworked I think. For instance google-repository.nix should be replaced with a fetchzip call.

File: pkgs/development/mobile/androidenv/build-gradle-app.nix [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    buildDir=`pwd`
+    cp -r $ANDROID_HOME $buildDir/local_sdk
+    chmod -R 755 local_sdk
+    export ANDROID_HOME=$buildDir/local_sdk
+    # Key files cannot be stored in the user's home directory. This
+    # overrides it.
+    export ANDROID_SDK_HOME=`pwd`
+
+    mkdir -p "$ANDROID_HOME/licenses"
+    echo -e "\n8933bad161af4178b1185d1a37fbf41ea5269c55" > "$ANDROID_HOME/licenses/android-sdk-license"
```
Comment (matthewbauer) [inline pkgs/development/mobile/androidenv/build-gradle-app.nix]: @ericson2314 Do you think this is okay to do? I guess the "assert acceptAndroidSdkLicenses" is basically covering this acceptance.

Comment (Ericson2314) [reply pkgs/development/mobile/androidenv/build-gradle-app.nix]: I...suppose so!

File: pkgs/development/mobile/androidenv/build-gradle-app.nix [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    cp -RL local_sdk/extras/google/m2repository/* .m2/repository/
+    gradle ${gradleTask} --offline --no-daemon -g ./tmp -Dmaven.repo.local=`pwd`/.m2/repository
+  '';
+
+  installPhase = ''
+    mkdir -p $out
+    mv ${buildDirectory}/build/outputs/apk/*.apk $out
+
+    mkdir -p $out/nix-support
+    echo "file binary-dist \"$(echo $out/*.apk)\"" > $out/nix-support/hydra-build-products
```
[the remark this answers is not in the document]
Comment (Ericson2314) [reply pkgs/development/mobile/androidenv/build-gradle-app.nix]: I don't know either. @ElvishJerricco might.

**Comment by ElvishJerricco** [comment]
As someone who helped write this stuff originally, I'm not really in favor of having this in nixpkgs in its current state. I never really considered it release quality. If possible, I'd *much* rather see this brought in-tree into reflex-platform.

**Comment by matthewbauer** [comment]
We can definitely look into removing this stuff but i would definitely object to adding even more android infrastructure to reflex-platform than there already is. This stuff is not specific to Reflex or even haskell so it does not seem like the right place to put it. We want more people to be using this outside of Reflex world so it gets more testing, more refactors, etc. It seems to me fairly close to doing what build-app.nix does for ant but with gradle. Maybe it makes sense to move all of this to some sort of a build support overlay or something like it though.

File: pkgs/development/mobile/androidenv/build-gradle-app.nix (hunk 1 above)
Comment (ElvishJerricco) [reply pkgs/development/mobile/androidenv/build-gradle-app.nix]: What is this hash?

File: pkgs/development/mobile/androidenv/build-gradle-app.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (ElvishJerricco) [reply pkgs/development/mobile/androidenv/build-gradle-app.nix]: No idea what this does

**Comment by ali-abrar** [comment]
Are there specific bugs or sources of instability you’re thinking of? What
features would you want it to have?

**Comment by dingxiangfei2009** [comment]
This PR does not have any tests to demonstrate the usability of the function "buildGradleApps". Shall we add some tests for it?

**Comment by ali-abrar** [comment]
This might be helpful: https://github.com/obsidiansystems/haven

On Fri, 28 Sep 2018 22:13:42 -0700
Ding Xiang Fei <notifications@github.com> wrote:

> This PR does not have any tests to demonstrate the usability of the function "buildGradleApps". Shall we add some tests for it?
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/NixOS/nixpkgs/pull/41855#issuecomment-425616806


-- 
Ali Abrar <aliabrar@gmail.com>

**Comment by Artturin** [comment]
Was accidentally removed in https://github.com/NixOS/nixpkgs/pull/50596

Readding in https://github.com/NixOS/nixpkgs/pull/254082

**Pull request merged**